dts/bindings/sensor/nxp,mcux-tempmon.yaml

```diff
@@ -0,0 +1,8 @@
+# Copyright (c) 2023 Centralp
+# SPDX-License-Identifier: Apache-2.0
+
+description: NXP MCUX temperature monitor
+
+compatible: "nxp,mcux-tempmon"
```

Comment:
another thing is let's call the binding "nxp,tempmon", we don't want soc family/marketing names in the IP binding names anymore and are trying to be rid of that